### PR TITLE
cargo,clippy,rustc,rustfmt: 1.76.0 -> 1.77.1

### PR DESCRIPTION
**Pull request by alyssais**
## Description of changes

Tested fd on aarch64-linux, and firefox, thunderbird, and fd on x86_64-linux.

~**Still looking for somebody to test fd on x86_64-darwin.**~

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by zowoq** [comment]
Build fd on x86_64-darwin.

**Comment by workingjubilee** [comment]
you may wish to be aware of https://github.com/NixOS/nixpkgs/issues/299606 but it is non-blocking I think, since it seems to require other variables to be involved in order to mess things up, and may not necessarily be considered a problem for ordinary use of nixpkgs.

also we're doing a point release tomorrow to fix a related bug, but only for Windows: https://github.com/rust-lang/rust/issues/122857 but maybe that matters for cross-compiling, idk.

**Comment by sdht0** [comment]
https://blog.rust-lang.org/2024/03/28/Rust-1.77.1.html is out

**Comment by winterqt** [comment]
Probably fine just to skip that point release, given it only affects MSVC?

**Review by blitz — APPROVED** [review]

**Comment by sdht0** [comment]
> Probably fine just to skip that point release, given it only affects MSVC?

Practically yes. OTOH repology will show the nixpkgs version to be outdated until the next release, for instance. Fine either ways.

**Review by phaer — APPROVED** [review]

File: pkgs/os-specific/linux/kernel/patches.nix [hunk 1]
```diff
@@ -88,4 +88,18 @@
...
+  rust_1_77-6_8 = {
+    name = "rust-1.77.patch";
+    patch = fetchurl {
+      name = "rust-1.77.patch";
+      url = "https://lore.kernel.org/rust-for-linux/20240217002717.57507-1-ojeda@kernel.org/raw";
+      hash = "sha256-0KW9nHpJeMSDssCPXWZbrN8kxq5bA434t+XuPfwslUc=";
+    };
+  };
+
+  rust_1_77-6_9 = {
```
Comment (Ma27) [inline pkgs/os-specific/linux/kernel/patches.nix:101]: I assume the patch is vendored because there's no stable location yet?

**Review by Ma27 — APPROVED** [review]
LGTM.

Requires a rebase, but other than that it looks good to me :+1:

File: pkgs/os-specific/linux/kernel/patches.nix (hunk 1 above)
Comment (alyssais) [reply pkgs/os-specific/linux/kernel/patches.nix:101]: Indeed.  Maybe now there is though, I'll check.

Comment (alyssais) [reply pkgs/os-specific/linux/kernel/patches.nix:101]: Yeah, it's been applied to rust-next, but [it looks like that gets force pushed sometimes](https://github.com/Rust-for-Linux/linux/activity?ref=rust-next), so there's probably no stable location until it's sent to Linus.

**Review by JohnRTitor — APPROVED** [review]
Can this be merged now?

**Pull request merged**